twitter_analytics_client.rb

```diff
@@ -0,0 +1,51 @@
+require 'mechanize'
+require 'rails'
+
+class TwitterAnalyticsClient
+  BASE_URI = 'https://twitter.com'.freeze
+  ANALYTICS_URI = 'https://analytics.twitter.com/user'.freeze
+
+  def initialize(user, password)
+    @user = user
+    @password = password
+  end
+
+  def login
+    @agent = Mechanize.new
+    page = @agent.get(BASE_URI)
+    form = page.forms[1]
+    form.field_with(name: "session[username_or_email]").value = @user
+    form.field_with(name: "session[password]").value = @password
+    form.submit
+    puts "Logged in Twitter @#{@user} password: #{@password}"
+  end
+
+  def get_analytics_data
+    return if @agent.nil?
+    @agent.post(export_url)
+    for i in 1..10 do
```

Comment:
[IMO] `i` isn't used. You can use `step` method. 

```ruby
10.step do
...
```

Reply:
I can't use `step` or `times` method because variable scope is limited when using these method.
I faced `undefined local variable or method "res" ` when using `step` and  `times`.
Ref: https://qiita.com/kanpou0108/items/8156c76a5723f4b48e3b

I searched the way  not to use unused variable but I can't find out.
So I use `for' now.

But please tell me if you have a good solution.

Reply:
> I can't use step or times method because variable scope is limited when using these method.

I didn't know that.

> So I use `for' now.

OK👍